src/parser/esprima_cpp/esprima.cpp

```diff
@@ -803,140 +809,111 @@ class Parser : public gc {
         }
     }
 
-    template <typename T>
-    PassRefPtr<Node> isolateCoverGrammar(T parseFunction)
+    void startCoverGrammar(IsolateCoverGrammarContext* grammarContext)
     {
-        const bool previousIsBindingElement = this->context->isBindingElement;
-        const bool previousIsAssignmentTarget = this->context->isAssignmentTarget;
-        Scanner::ScannerResult previousFirstCoverInitializedNameError = this->context->firstCoverInitializedNameError;
+        ASSERT(grammarContext != nullptr);
+
+        grammarContext->previousIsBindingElement = this->context->isBindingElement;
+        grammarContext->previousIsAssignmentTarget = this->context->isAssignmentTarget;
+        grammarContext->previousFirstCoverInitializedNameError = this->context->firstCoverInitializedNameError;
 
         this->context->isBindingElement = true;
         this->context->isAssignmentTarget = true;
         this->context->firstCoverInitializedNameError.reset();
 
-        this->checkRecursiveLimit();
-        PassRefPtr<Node> result = (this->*parseFunction)();
-        if (UNLIKELY(this->context->firstCoverInitializedNameError)) {
+        checkRecursiveLimit();
+    }
+
+    void endIsolateCoverGrammar(IsolateCoverGrammarContext* grammarContext)
+    {
+        ASSERT(grammarContext != nullptr);
+
+        if (this->context->firstCoverInitializedNameError) {
```

Comment:
please state expression explicitly like ``if (this->context->firstCoverInitializedNameError != nullptr)``

Reply:
This member is not a pointer anymore. It has a bool() cast operator though:

```
inline operator bool() const
{
    return this->type != InvalidToken;
}
```

I could add a `->type != InvalidToken` but I am not sure it looks better.

Reply:
Your suggention is more perpect in terms with expliciticity and readability! Please do like that. I promise you will get accustomed with new convention :)

Reply:
I did it. Shall I create another patch to remove all automated `bool` conversion?

Reply:
If you can, please do however don't be hurry it's not urgent.